### DISCUSSION TITLE
Dynamic header for my web app

### DISCUSSION THREAD

**Discussion by kehwar** [nuxt/nuxt#17421] [Questions]
I want to create a web app where the header displays the current page title or hides for some pages

However the header changes/toggles before the content of the new page is displayed which looks odd

I made this on the playground to illustrate the problem
https://stackblitz.com/edit/github-lkwxgs?file=README.md

I first used the route meta params, then switched to applying my config on the mounted hook, however it seems the mounted hook triggers before the component is actually displayed

**Comment by Flanker72** [comment]
Seems you have a problem with rendering a huge amount of data, this is a lag with rendering. This `v-for` rendering is a blocking operation. In real-world projects you should use a some kind of virtual lists to render such a huge amount of elements: refer [this article](https://blog.logrocket.com/rendering-large-datasets-vue-js/), [this article](https://blog.logrocket.com/create-performant-virtual-scrolling-list-vuejs/) and [this article](https://www.vuescript.com/best-virtual-scrolling/).

<img width="1679" alt="image" src="https://user-images.githubusercontent.com/619699/174483677-50b03e79-b7f5-4d81-8b47-32bb423e8bbd.png">
